TPP.Core/Commands/Definitions/OperatorCommands.cs

```diff
@@ -224,26 +218,18 @@ public async Task<CommandResult> AddRole(CommandContext context)
             (User user, Role role) = await context.ParseArgs<User, Role>();
             string response;
 
-            if (user.Roles != null)
+            HashSet<Role> roles = user.Roles;
+            bool roleAssigned = user.Roles.Add(role);
```

Comment:
I think you misunderstood me. Please don't modify the Roles set of user instances directly. I think it's fine to keep the actual type a HashSet for the reason that mongodb doesn't make it easy to use an actually immutable set, but the user object should still be treated immutable. Same goes for the Remove call further down.

It doesn't make much of a difference for this case, so it's not a big deal. If there's a good reason to mutate it in-place, you may keep it

Reply:
This was a typo, I meant to modify the copy "roles" just above but... didn't.

Reply:
the line above it isn't a copy though

Reply:
you have a point. That will be changed.